lib/codegen/CreateStaticTerm.cpp

```diff
@@ -0,0 +1,318 @@
+#include "kllvm/codegen/CreateStaticTerm.h"
+#include "kllvm/codegen/Debug.h"
+#include "kllvm/codegen/Util.h"
+
+#include <iomanip>
+#include <sstream>
+
+#include "runtime/header.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/DerivedTypes.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IRReader/IRReader.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace kllvm {
+
+extern std::string escape(std::string str);
+
+/* create a term, given the assumption that the created term will not be a
+ * triangle injection pair */
+llvm::Constant *CreateStaticTerm::notInjectionCase(
+    KORECompositePattern *constructor, llvm::Constant *val) {
+  const KORESymbol *symbol = constructor->getConstructor();
+  llvm::StructType *BlockType = getBlockType(Module, Definition, symbol);
+
+  std::stringstream koreString;
+  constructor->print(koreString);
+  llvm::Constant *Block
+      = Module->getOrInsertGlobal(koreString.str().c_str(), BlockType);
+  llvm::GlobalVariable *globalVar = llvm::dyn_cast<llvm::GlobalVariable>(Block);
+
+  if (!globalVar->hasInitializer()) {
+    std::vector<llvm::Constant *> blockVals;
+
+    llvm::StructType *BlockHeaderType
+        = getTypeByName(Module, BLOCKHEADER_STRUCT);
+    uint64_t headerVal = symbol->getTag();
+    uint64_t sizeInBytes = llvm::DataLayout(Module).getTypeAllocSize(BlockType);
+    assert(sizeInBytes % 8 == 0);
+    headerVal |= (sizeInBytes / 8) << 32;
+    headerVal |= (uint64_t)symbol->getLayout() << LAYOUT_OFFSET;
```

Comment:
Can we please refactor this out into its own function rather than duplicating the logic in `getBlockHeader`?